nnvm/python/nnvm/frontend/tensorflow.py

```diff
@@ -1177,43 +1183,65 @@ def from_tensorflow(self, graph, layout="NHWC", shape=None, outputs=None):
             raise NotImplementedError( \
                 "The following operators are not implemented: {}".format(missing_operators))
 
+        for node in graph.node:
+            if node.op == 'Placeholder':
+                self._input_shapes[node.name] = \
+                    tensor_util.TensorShapeProtoToList(node.attr['shape'].shape)
+
+                if shape and node.name in shape:
+                    for idx, dim in enumerate(self._input_shapes[node.name]):
```

Comment:
Just pickup the complete shape from user input instead of only some dimensions.

Reply:
` if dim < 0:
self._input_shapes[node.name][idx] = shape[node.name][idx]` is for corner case when N is -1, then user's N will be used instead of -1, keep the same for all the other dimensions. The assert will help check user's input shape, such as for const

Reply:
It makes sense to me, thanks for the comment. Will update the code.